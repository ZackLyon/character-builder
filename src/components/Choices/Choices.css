.choices {
  display: flex;
  flex-direction: column;
}

select {
  margin-left: 1vw;
}
