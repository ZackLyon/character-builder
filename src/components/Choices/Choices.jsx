import React from 'react';

export default function Choices({ head, setHead, middle, setMiddle }) {
  const heads = ['angry', 'cracked', 'crazy', 'happy', 'neutral'];
  const middles = ['blue', 'fancy', 'pink', 'red'];

  return (
    <div>
      <label>
        Heads:
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          {heads.map((headOption) => (
            <option key={headOption}>{headOption}</option>
          ))}
        </select>
      </label>

      <label>
        Middles:
        <select value={middle} onChange={(e) => setMiddle(e.target.value)}>
          {middles.map((middleOption) => (
            <option key={middleOption}>{middleOption}</option>
          ))}
        </select>
      </label>
    </div>
  );
}
