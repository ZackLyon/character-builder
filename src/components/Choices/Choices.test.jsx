import { render, screen } from '@testing-library/react';
import Choices from './Choices';

it('should display the choices on the screen', () => {
  const { container } = render(<Choices />);

  const heads = screen.getByLabelText('Heads:');
  const middles = screen.getByLabelText('Middles:');
  const legs = screen.getByLabelText('Legs:');
  const catchphrase = screen.getByLabelText('Catch Phrase:');

  expect(heads).toBeInTheDocument();
  expect(middles).toBeInTheDocument();
  expect(legs).toBeInTheDocument();
  expect(catchphrase).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
