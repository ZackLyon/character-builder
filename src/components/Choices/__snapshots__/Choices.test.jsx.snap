// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the choices on the screen 1`] = `
<div>
  <div
    class="choices"
  >
    <label>
      Heads:
      <select>
        <option>
          angry
        </option>
        <option>
          cracked
        </option>
        <option>
          crazy
        </option>
        <option>
          happy
        </option>
        <option>
          neutral
        </option>
      </select>
    </label>
    <label>
      Middles:
      <select>
        <option>
          blue
        </option>
        <option>
          fancy
        </option>
        <option>
          pink
        </option>
        <option>
          red
        </option>
      </select>
    </label>
    <label>
      Legs:
      <select>
        <option>
          blue
        </option>
        <option>
          dog
        </option>
        <option>
          leg
        </option>
        <option>
          white
        </option>
      </select>
    </label>
    <label>
      Catch Phrase:
      <input
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Submit
      </button>
    </label>
  </div>
</div>
`;
